roles/servicetelemetry/tasks/component_qdr.yml

```diff
@@ -160,21 +160,40 @@
         namespace: "{{ ansible_operator_meta.namespace }}"
       register: _qdr_basicauth_object
 
-    # Because https://github.com/interconnectedcloud/qdr-operator/blob/576d2b33dac71437ea2b165caaaf6413220767fe/pkg/controller/interconnect/interconnect_controller.go#L634
-    - name: Perform a one-time upgrade to the default generated password for QDR BasicAuth
-      k8s:
-        definition:
-          kind: Secret
-          apiVersion: v1
-          metadata:
-            name: "{{ ansible_operator_meta.name }}-interconnect-users"
+    - when:
+      - _qdr_basicauth_object.resources[0] is defined and _qdr_basicauth_object.resources[0].metadata.labels.stf_one_time_upgrade is not defined
+      block:
+        # Because https://github.com/interconnectedcloud/qdr-operator/blob/576d2b33dac71437ea2b165caaaf6413220767fe/pkg/controller/interconnect/interconnect_controller.go#L634
+        - name: Perform a one-time upgrade to the default generated password for QDR BasicAuth
+          k8s:
+            definition:
+              kind: Secret
+              apiVersion: v1
+              metadata:
+                name: "{{ ansible_operator_meta.name }}-interconnect-users"
+                namespace: "{{ ansible_operator_meta.namespace }}"
+                labels:
+                  stf_one_time_upgrade: "{{ lookup('pipe', 'date +%s') }}"
+              stringData:
+                guest: "{{ lookup('password', '/dev/null chars=ascii_letters,digits length=32') }}"
+
+        - name: Get name of QDR pod (label_selectors on the k8s object need kubernetes.core>=2.2.0)
+          k8s_info:
+            api_version: v1
+            kind: Pod
+            namespace: "{{ ansible_operator_meta.namespace }}"
+            label_selectors:
+              - application={{ ansible_operator_meta.name }}-interconnect
+          register: _qdr_pod
+
+        - name: Restart QDR pod to pick up new password
+          k8s:
+            state: absent
+            api_version: v1
+            kind: Pod
             namespace: "{{ ansible_operator_meta.namespace }}"
-            labels:
-              stf_one_time_upgrade: "{{ lookup('pipe', 'date +%s') }}"
-          stringData:
-            guest: "{{ lookup('password', '/dev/null chars=ascii_letters,digits length=32') }}"
-      when:
-        - _qdr_basicauth_object.resources[0] is defined and _qdr_basicauth_object.resources[0].metadata.labels.stf_one_time_upgrade is not defined
+            name: "{{ _qdr_pod.resources[0].metadata.name }}"
```

Comment:
While deprecated, what happens in an HA environment when there are multiple pods? (I suspect only 1 will be restarted.)

I can't wait to remove HA in 1.5.4 ;)

Reply:
Damn. This did cross my mind after I posted the PR (so I went to double-check HA was officially deprecated) and you had to go and comment on it! ;)  I can add a loop and test HA.

Reply:
Alternatively, we could just not support auth mode when HA mode is enabled?

I've been wanting to figure out Events in k8s for a while now so we could write these things into the console UI etc. A bit much for right now, but there could be a check that doesn't allow highAvailability.enabled=true + qdr.auth=basic to be configured at the same time?

Reply:
TBH though, thinking about writing notes in the documentation stating these items aren't supported together, adding a validation code set to check for the configuration state, and updating the release notes for the QDR basic auth to state it won't work with HA mode, is probably just more work than adding the loop system and running a test.